Calendar

Zacchua opened the pull request:
Added calendar with basic functionality. 
- select date shows the events for the day below the calendar.
- days with events have dots under them

Zacchua committed:
Added calendar and dummy card

one simple card with dummy data,

Zacchua committed:
Merge branch 'main' into calendar

Zacchua committed:
Added support for list of cards

Zacchua committed:
Merge branch 'main' into calendar

Zacchua committed:
Added support for filtering cards by day

Zacchua committed:
Markings on calendar for events and selected date

Also added update event on tap functionality.

Zacchua committed:
Merge branch 'main' into calendar

Zacchua committed:
Fixed formatting issues

kevin9foong commented on src/screens/home/CalendarScreen.tsx:
```diff
@@ -1,8 +1,116 @@
+		return card.startdate.toDateString() === date.toDateString() && card.duedate.toDateString() === date.toDateString();
+	};
+	
+	const getDayText = (num: number): string => {
```
hi zach, could you abstract these date formatting code into a separate utils/dateFormat file? so that it can be used throughout the project?

kevin9foong commented on src/components/home/CalendarCard.tsx:
```diff
@@ -0,0 +1,39 @@
+
+	const item = info.item;
+
+	const formatMinutes = (mins: number): string => {
```
these as well - abstract them to the same utils file as the below number -> day mapper

kevin9foong commented on src/components/home/CalendarCard.tsx:
```diff
@@ -0,0 +1,39 @@
+		return mins === 0 ? '00' : mins.toString();
+	};
+  
+	const getTimeRange = (card: CardModelWithUid): string => {
```
this as well, can be used for the main component so would be good to be abstracted

Zacchua committed:
Extracted utility functions into a separate file

Also have been getting a red line under react-native-calendars import in
calendar screen so i tried to add it again using yarn but it didnt
remove the red line. Functionality of the calendar is not affected
though so i just left it

Zacchua committed:
Added support for multi day events in the calendar

Multiday events are displayed as highlighted dates, the card shown has
the start and end date displayed instead of the timings (as compared to
single day events)

Zacchua committed:
Added support for multi day events

Previously color of the event line relied solely on the number of lines
on the first day. Fixed it such that it checks the duration of the event
for other events falling on the same day.

Zacchua committed:
Merge branch 'main' into calendar

Zacchua committed:
Fixed bug that prevent some tasks from being shown

Minor bug in formatting of the date string that caused the multi period
marking to break for single digit dates eg. 3/07/2021. Also fixed corner
case where card for multi day event would not be shown for the first day

kevin9foong committed:
Add rn testing libraries

Zacchua committed:
Merge branch 'main' into calendar

Zacchua committed:
Improved styling for current selected day

current day is now represented as a dark blue circle instead and does
not override other styling on that date.

Could not get agenda to work though. It just didnt show up

Zacchua committed:
Merge branch 'calendar' of https://github.com/kevin9foong/prodpets into calendar

Zacchua committed:
fixed weird netinfo error

Zacchua committed:
Added some simple initial tests

Zacchua committed:
Added test for Calendar screen with mock data

Zacchua committed:
Modified tests abit

Added tests for title of Calendar cards

Zacchua committed:
Removed unused function

also testing out new ssh keys

Zacchua committed:
Merge branch 'main' into calendar

kevin9foong committed:
Merge branch 'calendar' of github.com:kevin9foong/prodpets into calendar

# Conflicts:
#	yarn.lock

kevin9foong committed:
Install unmet peer dependeneces

kevin9foong committed:
Update to Expo v42

kevin9foong committed:
Bump up versions of redux-offline and react navigation

kevin9foong committed:
Rename variables, merge view-modal, useTime

Zacchua committed:
Merge branch 'main' into calendar

Zacchua committed:
Refactored Calendar to Agenda
